interfaces/repo.go

```diff
@@ -80,6 +80,112 @@ func (r *Repository) AddInterface(i Interface) error {
 	return nil
 }
 
+// QueryOptions describes options for QueryInterfaces.
+//
+// Names: return just this subset if non-empty.
+// Doc: return documentation.
+// Plugs: return information about plugs.
+// Slots: return information about slots.
+// Connected: only consider interfaces with at least one connection.
+type QueryOptions struct {
```

Comment:
One solution might be to organize that new functionality in terms of "information about interfaces", which is something we don't have yet. For that, we can call this type "InfoOptions".